src/MoBi.Presentation/Tasks/SimulationCommitTask.cs

```diff
@@ -0,0 +1,126 @@
+      private IMoBiCommand addNewParameterValuesFromSimulationChanges(IMoBiSimulation simulationWithChanges, Module lastModule)
+      {
+         var parameterValuesToAdd = changesFrom<Parameter>(simulationWithChanges).Select(x => createParameterValue(x.quantity, x.objectPath)).ToList();
+         return createAddBuildingBlockCommand<ParameterValuesBuildingBlock, ParameterValue>(simulationWithChanges.Name, lastModule.ParameterValuesCollection.AllNames(), lastModule, parameterValuesToAdd);
+      }
+
+      private IMoBiCommand addNewInitialConditionsFromSimulationChanges(IMoBiSimulation simulationWithChanges, Module lastModule)
+      {
+         var initialConditionsToAdd = changesFrom<MoleculeAmount>(simulationWithChanges).Select(x => createInitialCondition(x.quantity, x.objectPath)).ToList();
+         return createAddBuildingBlockCommand<InitialConditionsBuildingBlock, InitialCondition>(simulationWithChanges.Name, lastModule.InitialConditionsCollection.AllNames(), lastModule, initialConditionsToAdd);
+      }
+
+      /// <summary>
+      ///    Returns path and quantity tuple for all changes in the <paramref name="simulationWithChanges" /> of type
+      ///    <typeparamref name="TQuantity" />
+      /// </summary>
+      private IEnumerable<(ObjectPath objectPath, TQuantity quantity)> changesFrom<TQuantity>(IMoBiSimulation simulationWithChanges) where TQuantity : Quantity
+      {
+         var quantities = _entitiesInSimulationRetriever.EntitiesFrom<TQuantity>(simulationWithChanges);
+         return simulationWithChanges.OriginalQuantityValues.Where(x => quantities.Contains(x.Path)).Select(x => (objectPath: x.Path, quantity: quantities[x.Path]));
+      }
+
+      private IMoBiCommand createAddBuildingBlockCommand<TBuildingBlock, TPathAndValueEntity>(string buildingBlockName, IEnumerable<string> usedNames, Module moduleToAddTo, IReadOnlyList<TPathAndValueEntity> entitiesToAdd)
+         where TBuildingBlock : PathAndValueEntityBuildingBlock<TPathAndValueEntity>, IBuildingBlock where TPathAndValueEntity : PathAndValueEntity
+      {
+         if (!entitiesToAdd.Any())
+            return new MoBiEmptyCommand();
+
+         var buildingBlock = _context.Create<TBuildingBlock>().WithName(buildingBlockName);
+         entitiesToAdd.Each(buildingBlock.Add);
+         _nameCorrector.AutoCorrectName(usedNames, buildingBlock);
+         return new AddBuildingBlockToModuleCommand<TBuildingBlock>(buildingBlock, moduleToAddTo);
+      }
+
+      private InitialCondition createInitialCondition(MoleculeAmount moleculeAmount, ObjectPath path)
+      {
+         var newPath = path.Clone<ObjectPath>();
+         // Remove the name from the path before creating the initial condition with separate arguments for name and path
+         newPath.RemoveAt(newPath.Count - 1);
+         var initialCondition = _initialConditionsCreator.CreateInitialCondition(newPath, path.Last(), moleculeAmount.Dimension, moleculeAmount.DisplayUnit, moleculeAmount.ValueOrigin);
```

Comment:
feels like the interface here is missing a property lol. CreateIniticalCOndiition(path, moleculeAmouhnt) like we have for parameter. This is not symmetrical and it's ugly :)